x/app-router/keeper/liquidstake.go

```diff
@@ -0,0 +1,68 @@
+package keeper
+
+import (
+	"github.com/armon/go-metrics"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/v3/modules/core/exported"
+
+	"github.com/Stride-Labs/stride/v3/x/app-router/types"
+	stakeibckeeper "github.com/Stride-Labs/stride/v3/x/stakeibc/keeper"
+	stakeibctypes "github.com/Stride-Labs/stride/v3/x/stakeibc/types"
+)
+
+func (k Keeper) TryLiquidStaking(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	newData transfertypes.FungibleTokenPacketData,
+	parsedReceiver *types.ParsedReceiver,
+	ack ibcexported.Acknowledgement,
+) ibcexported.Acknowledgement {
+	params := k.GetParams(ctx)
+	if !params.Active {
+		return channeltypes.NewErrorAcknowledgement("packet forwarding param is not active")
+	}
+
+	// In this case, we can't process a liquid staking transaction, because we're dealing with STRD tokens
+	if transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), newData.Denom) {
+		return channeltypes.NewErrorAcknowledgement("the native token is not supported for liquid staking")
+	}
+
+	amount, ok := sdk.NewIntFromString(newData.Amount)
+	if !ok {
+		return channeltypes.NewErrorAcknowledgement("not a parsable amount field")
+	}
+
+	// Note: newData.denom is base denom e.g. uatom - not ibc/xxx
+	var token = sdk.NewCoin(newData.Denom, amount)
+
+	err := k.RunLiquidStake(ctx, parsedReceiver.StrideAccAddress, token, []metrics.Label{})
+	if err != nil {
+		ack = channeltypes.NewErrorAcknowledgement(err.Error())
+	}
+	return ack
+}
+
+func (k Keeper) RunLiquidStake(ctx sdk.Context, addr sdk.AccAddress, token sdk.Coin, labels []metrics.Label) error {
+	msg := &stakeibctypes.MsgLiquidStake{
+		Creator:   addr.String(),
```

Comment:
How does signing work for this transaction? Does broadcasting a `LiquidStake`(`MsgLiquidStake`) with `Creator=receiverAddress` require the receiver to sign the tx from their Stride address, or is their hostZone account's signature (when sending the tx) sufficient?

Reply:
Just hostZone account's signature is sufficient to run this.